doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,94 @@
+---
+author: <Pankaj> <Bhojwani> <pabhojwa@microsoft.com>
+created on: <2020-11-20>
+last updated: <2020-11-20>
+issue id: <#8345>
+---
+
+# Configuration object for profiles
+
+## Abstract
+
+This spec outlines how we can support 'configuration objects' in our profiles, which
+will allow us to render differently depending on the state of the control. For example, a
+control can be rendered differently if its focused as compared to when its unfocused. Another
+example is that an elevated state control can be rendered differently as compared to a
+non-elevated one.
+
+## Inspiration
+
+Reference: [#3062](https://github.com/microsoft/terminal/issues/3062)
+
+Users want there to be a more visible indicator than the one we have currently for which
+pane is focused and which panes are unfocused. This change would grant us that feature.
+
+## Solution Design
+
+A new class in the `TerminalControl` namespace, called `CustomRenderConfig`, that will contain rendering
+parameters. `TermControl` and `TerminalCore` will use objects of this class to pass along rendering parameters
+to the renderer.
+
+Our `TerminalSettingsModel` will parse out that object from the settings json file and pipe it over to
+`TermControl`/`TerminalCore` to use. This will be done through `IControlSettings` and `ICoreSettings`, which
+is the way we already pipe over information that these interfaces need to know regarding rendering (such as
+`CursorStyle` and `FontFace`).
+
+### Allowed parameters
+
+Not all parameters which can be defined in a `Profile` can be defined in this new object (for example, we
+do not want parameters which would cause a resize in this object.) Here are the list of parameters we
+will allow:
+
+- Anything regarding colors: `colorScheme`, `foreground`, `background`, `cursorColor` etc
+- `cursorShape`
+- `backgroundImage`
+
+## UI/UX Design
+
+Users will be able to add a new setting to their profiles that will look like this:
+
+```
+"unfocusedState": 
+{
+    "colorScheme": "Campbell",
+    "cursorColor": "#888",
+    "cursorShape": "emptyBox",
+    "foreground": "#C0C0C0",
+    "background": "#000000"
+}
+```
+
+## Capabilities
+
+### Accessibility
+
+Does not affect accessibility.
+
+### Security
+
+Does not affect security.
+
+### Reliability
+
+This is another location in the settings where parsing/loading the settings may fail. However, this is the case
+for any new setting we add so I would say that this is a reasonable cost for this feature.
+
+### Compatibility
+
+Should not affect compatibility.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues

```

Comment:
We 100% need to mention `profile.defaults` here (or somewhere). How do we handle variants of this case, where some of the fontFace's are missing:
```js
"profiles": {
  "defaults": {
    "fontFace": "PD_Standard",
    "unfocusedState": {
      "fontFace": "PD_Unfocused"
    }
  },
  "list": [
    {
      "fontFace": "Standard",
      "unfocusedState": {
        "fontFace": "Unfocused"
      }
    }
  ]
}
```

The question is, how does inheritance work:
1. **Standard**: each _setting_ is conditionally applied (is in state?)
  a. unfocusedState --> Profile --> PD.unfocusedState --> PD --> System setting
2. **All-or-nothing**: each _state_ is conditionally applied (is in state?)
  a. unfocusedState --> Profile --> PD --> System setting
  b. (unfocusedState is empty) --> Profile --> PD.unfocusedState --> PD --> System setting
I'm leaning towards 2, I think? But I'm still unsure.

Reply:
This is definitely something we need to discuss (I mentioned it in the body of #8392 but forgot to put it here as well)

Reply:
Resolved - going for something similar to option 2